### ISSUE TITLE
Consider collapsing domain options when you select one

### ISSUE THREAD

**Issue by jasmussen** [Automattic/wp-calypso#98667]
### What

This GIF shows the flow of using the keyboard to select the free plan, and then tab forward to "Continue":

![Image](https://github.com/user-attachments/assets/899cb6b2-7782-4c9f-9237-5bd89a441e7f)

_While there's a separate missing-focus-style issue, this issue is focused on the specific behavior of selecting a domain option._ 

What happens is: you click a domain option, the button you clicked changes to say "✅ Selected", and you can proceed to click the Continue button on the right.

This flow works, but it comes with a couple of downsides:

* Once I've selected a plan, the other plans are likely to be less relevant to me. I've made my choice.
* If I'm a keyboard user, I can't just press "Tab" once to focus the Continue button. I have to tab through every other option on the page first, which is 11 tabstops. If you're also using a screenreader, this gets very verbose.

The opportunity here is to improve the flow overall, for both clarity and keyboarding.

One idea is to collapse every option but the selected one when clicking the plan. This would remove all other plan tab-stops from the flow, and make the next next tab-stop the "Remove" button, and then "Continue" after that. 

This would need some design consideration, to ensure how the back and forth would work. For example, the "Remove" button would presumably uncollapse the options again.

Further context: p58i-jbA-p2

### Why

This will benefit flow across keyboard and mouse.

### How

Through finding a design that works for both.
